src/bunit.web/Extensions/InputFile/InputFileExtensions.cs

```diff
@@ -16,13 +16,30 @@ public static class InputFileExtensions
 	public static void UploadFiles(
 		this IRenderedComponent<InputFile> inputFileComponent,
 		params InputFileContent[] files)
+		=> UploadFilesAsync(inputFileComponent, files).GetAwaiter().GetResult();
```

Comment:
```suggestion
		{
			var uploadTask = UploadFilesAsync(inputFileComponent, files);
			if(!uploadTask.IsCompleted)
			{
				throw new SomeGoodExceptionName("Upload did not complete synchronously. Use `UploadFilesAsync` instead.");
			}
		}
```

This is just hacked together here, but I kind of like this pattern instead of using GetAwaiter().GetResult(). We are basically warning users that they their upload did not complete, and they should use the async version. This makes the change very discoverable, and those that don't need to use the async version, they can continue to use the other version without problems.

Reply:
Good point. At least it's better than my current anti-pattern *sync over async* to just call `GetAwaiter().GetResult()`.

Still it feels a bit strange, that the sync version doe semantically something else than the async version, not?
Anyway let's give it a shot

Reply:
Yeah, it's a bit weird, but a good middle ground, and I prefer it to use the obsolete attribute. The key is to make the exception message very clear and obvious.